packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/ListView/index.js

```diff
@@ -45,212 +42,156 @@ import {
   VisuallyHidden,
 } from '@strapi/design-system';
 import { Plus, Pencil, Trash, EmptyDocuments } from '@strapi/icons';
-import reducer, { initialState } from './reducer';
 import adminPermissions from '../../../../../permissions';
 
 const ListView = () => {
-  const {
-    isLoading,
-    allowedActions: { canCreate, canRead, canUpdate, canDelete },
-  } = useRBAC(adminPermissions.settings.webhooks);
-  const toggleNotification = useNotification();
-  const isMounted = useRef(true);
-  const { formatMessage } = useIntl();
+  const [webhooks, setWebhooks] = useState([]);
+  const [webhookToDelete, setWebhookToDelete] = useState(null);
+  const [webhooksToDelete, setWebhooksToDelete] = useState([]);
   const [showModal, setShowModal] = useState(false);
-  const [{ webhooks, webhooksToDelete, webhookToDelete, loadingWebhooks }, dispatch] = useReducer(
-    reducer,
-    initialState
-  );
-  const { notifyStatus } = useNotifyAT();
-
-  const { get, del, post, put } = useFetchClient();
 
+  const { formatMessage } = useIntl();
+  const { formatAPIError } = useAPIErrorHandler();
+  const toggleNotification = useNotification();
   useFocusWhenNavigate();
   const { push } = useHistory();
   const { pathname } = useLocation();
+
+  const {
+    isLoading: isRBACLoading,
+    allowedActions: { canCreate, canUpdate, canDelete },
+  } = useRBAC(adminPermissions.settings.webhooks);
+  const { get, del, post, put } = useFetchClient();
+  const { notifyStatus } = useNotifyAT();
+
   const rowsCount = webhooks.length;
   const webhooksToDeleteLength = webhooksToDelete.length;
   const getWebhookIndex = (id) => webhooks.findIndex((webhook) => webhook.id === id);
 
-  useEffect(() => {
-    isMounted.current = true;
-
-    return () => {
-      isMounted.current = false;
-    };
-  }, []);
-
-  /**
-   * TODO: refactor this, but actually refactor
-   * the whole component. Needs some love.
-   */
-  useEffect(() => {
-    const fetchWebHooks = async () => {
-      try {
-        const {
-          data: { data },
-        } = await get('/admin/webhooks');
+  const fetchWebHooks = async () => {
```

Comment:
I think the way this function is written isn't utilising react-query to the best it could be, realistically the query function can be inlined and as simple as:

```js
const { isLoading: webhooksLoading, data: webhooks } = useQuery(QUERY_KEY, async () => {
  const { data: { data } } = await get('/admin/webhooks');
  return data
})
```

You can then use `useEffect` to trigger notifications on `fail` or `success` – I wouldn't use the callbacks because they will only fire when new data is successfully fetched / cache is updated as opposed to hitting the cache.